### PR TITLE
py-tensorflow: version updates, add 2.4.0

### PR DESCRIPTION
**Pull request by aweits**

### CODE REVIEW

**Review by adamjstewart — COMMENTED** [review]
Ugh, they moved to using `~=` instead of `>=` for Python dependencies. For example, `astunparse ~= 1.6.3` means `py-astunparse@1.6.3:1.6.999`. Can you update these?

**Comment by aweits** [comment]
> Ugh, they moved to using `~=` instead of `>=` for Python dependencies. For example, `astunparse ~= 1.6.3` means `py-astunparse@1.6.3:1.6.999`. Can you update these?

Yep, I have to go over a bunch of things here still - it's probably time to get rid of "overlapping" things like this as well:
```
depends_on('py-numpy@1.13.3:',       type=('build', 'run'), when='@1.6:')
depends_on('py-numpy@1.12.1:',       type=('build', 'run'), when='@1.4:')
depends_on('py-numpy@1.11.0:',       type=('build', 'run'), when='@0.11:')
depends_on('py-numpy@1.10.1:',       type=('build', 'run'), when='@0.7.1: platform=darwin')
depends_on('py-numpy@1.8.2:',        type=('build', 'run'), when='@0.6:')
```

**Comment by aweits** [comment]
I've built/tested 1.15.5, 2.0.4, 2.1.3, 2.2.2, 2.3.2, and 2.4.0 at this point.

File: var/spack/repos/builtin/packages/py-tensorflow/package.py [hunk 1]
```diff
@@ -138,54 +149,64 @@ class PyTensorflow(Package, CudaPackage):
...
+    depends_on('py-wheel@0.35:0.999', type=('build', 'run'), when='@2.4.0: ^python@3:')
     depends_on('py-mock@2.0.0:', type=('build', 'run'), when='@0.10: ^python@:2')
     depends_on('py-functools32@3.2.3:', type=('build', 'run'), when='@1.15: ^python@:2')
-    depends_on('py-six@1.12.0:', type=('build', 'run'), when='@2.1:')
+    depends_on('py-six@1.15.0:1.15.999', type=('build', 'run'), when='@2.4.0:')
+    depends_on('py-six@1.12.0:', type=('build', 'run'), when='@2.1:2.3')
     depends_on('py-six@1.10.0:', type=('build', 'run'), when='@:2.0')
     depends_on('py-scipy@1.2.2', type=('build', 'run'), when='@2.1.0:2.1.1,2.2.0,2.3.0 ^python@:2')
     depends_on('py-scipy@1.4.1', type=('build', 'run'), when='@2.1.0:2.1.1,2.2.0,2.3.0 ^python@3:')
     depends_on('py-grpcio@1.8.6:', type=('build', 'run'), when='@1.6:1.7')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tensorflow/package.py:208]: You missed one:
```python
# grpcio does not build correctly on big-endian machines due to lack of         
# BoringSSL support.                                                            
# See https://github.com/tensorflow/tensorflow/issues/17882.                    
if sys.byteorder == 'little':                                                   
  REQUIRED_PACKAGES.append('grpcio ~= 1.32.0')
```

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-tensorflow/package.py:208]: We also might want to wrap the `depends_on` in the same if-statement

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**